scipy/optimize/_qap.py

```diff
@@ -186,10 +200,36 @@ def quadratic_assignment(A, B, method="faq", options=None):
                "2opt": _quadratic_assignment_2opt}
     if method not in methods:
         raise ValueError(f"method {method} must be in {methods}.")
+
+    _spec007_transition(options.get("rng", None))
     res = methods[method](A, B, **options)
     return res
 
 
+def _spec007_transition(rng):
+    if isinstance(rng, np.random.RandomState):
+        warnings.warn(
+            "Use of `RandomState` with `quadratic_assignment`is being phased out"
+            " and will result in an Exception in SciPy 1.17",
+            DeprecationWarning,
+            stacklevel=2
+        )
+    if rng is None and np.random.mtrand._rand._bit_generator._seed_seq is None:
+        warnings.warn(
+            "The NumPy global RNG was seeded by calling `np.random.seed`."
+            " From SciPy 1.17, this function will no longer use the global RNG.",
+            FutureWarning,
+            stacklevel=2
+        )
+    if isinstance(rng, numbers.Integral | np.integer):
+        warnings.warn(
+            "From SciPy 1.17, use of an integer seed will be normalized via"
+            " `np.random.default_rng`, rather than `np.random.RandomState`",
```

Comment:
To avoid having to wordsmith, something like this would more closely follows the SPEC wording:

> The behavior when the `rng` option is an integer is changing: the value will be normalized using `np.random.default_rng` beginning in SciPy 1.17, and the resulting `Generator` will be used to generate random numbers.